server/api/studio/studio.go

```diff
@@ -70,6 +71,22 @@ func main() {
 	// api handlers
 	handler.RegisterHandlers(server, svcCtx)
 
+	// websocket
+	hub := ws.NewHub()
+	go hub.Run()
+	server.AddRoute(rest.Route{
+		Method: http.MethodGet,
+		Path:   "/nebula_ws",
+		Handler: func(w http.ResponseWriter, r *http.Request) {
+			clientInfo := &auth.AuthData{}
+			tokenCookie, err := r.Cookie(svcCtx.Config.Auth.TokenName)
+			if err == nil {
+				clientInfo, _ = auth.Decode(tokenCookie.Value, svcCtx.Config.Auth.AccessSecret)
+			}
```

Comment:
if err != nil, need handle err?

Reply:
should make sure websocket connecting succeed